editor/components/post-publish-panel/prepublish.js

```diff
@@ -11,6 +11,7 @@ import PostVisibility from '../post-visibility';
 import PostVisibilityLabel from '../post-visibility/label';
 import PostSchedule from '../post-schedule';
 import PostScheduleLabel from '../post-schedule/label';
+import PluginPrePublishPanel from '../../../edit-post/components/plugin-pre-publish-panel';
```

Comment:
`editor` module shouldn't depend on `edit-post`.
@youknowriad, should we move the PublishPanel to `edit-post`?

Reply:
What's the high-level difference between which content is in `edit-post` vs `editor`?

Reply:
`editor` is a generic module for building any editor layout
`edit-post` is the layout built for editing posts/pages/cpts
we can imagine `edit-p2` `edit-template`...

A simple solution here would be to use the  `children` prop.

Reply:
>editor is a generic module for building any editor layout
edit-post is the layout built for editing posts/pages/cpts
we can imagine edit-p2 edit-template...

Aha, I see. It makes sense to me now why we can't have `editor` depending on `edit-post`.

>A simple solution here would be to use the children prop.

Are you suggesting including `PluginPrePublishPanel` in the `edit-post` layout (/edit-post/component/layout/index.js). Maybe something like the following? --
```jsx
			{ publishSidebarOpened && (
				<PostPublishPanel
					onClose={ closePublishSidebar }
					forceIsDirty={ hasActiveMetaboxes }
					forceIsSaving={ isSaving }
				>
					<PluginPrePublishPanel.Slot/>
				</PostPublishPanel>
			) }
```

And then of course, passing it on the way down so it gets included in the right place in the `editor` layout?

That would keep `editor` as generic as possible. However, I could see the advantage of including the `PluginPrePublishPanel` Slot in `editor`. I'm guessing this extensibility would be useful for any `editor` implementation.

Reply:
> That being said, I could see the advantage of including the PluginPrePublishPanel Slot in editor. I'm guessing this extensibility would be useful for any editor implementation.

I don't think we should force an extensibility pattern like this to all editors. Saving a template is not the same as saving a post. I'd prefer the former (like the example you shown)

Reply:
Yes, `children` seems like the way to go 👍

Reply:
@gziolo, I incorporated the new createSlotFill! Thanks!

I also made the change to remove the `edit-post` dependency in `editor`. I couldn't use strictly `children` since the top level `editor` component `<PostPublishPanel>` contains both the 'pre' and the 'post' components. I created `prePublishExtension` and `postPublishExtension` for this purpose instead to route the plugin slots down to right places.